### PR TITLE
NCMesh: fix 8-bit reference-counter overflow at high-valence vertices

### PR DESCRIPTION
**Pull request by Sbozzolo**
NCMesh::Node stores per-node vertex/edge reference counts in a `char`. ReferenceElement() increments vert_refc/edge_refc once per referencing element, so the count equals the vertex/edge valence in the coarse mesh. For a vertex whose valence is a multiple of 256 (unsigned char, e.g. ARM / Graviton) or >= 128 (signed char, e.g. x86), the counter overflows.

When vert_refc wraps to a non-positive value, Node::HasVertex() returns false, so NCMesh::UpdateVertices() never records the node id in vertex_nodeId[] for that vertex; it keeps its default value 0. Later, NCMesh::OnMeshUpdated() looks the node up via
nodes.Find(vertex_nodeId[ev[0]], vertex_nodeId[ev[1]]) and dereferences the result. With the corrupted mapping the lookup returns null:

  - debug builds:   MFEM_ASSERT fires -> "edge (...) not found"
  - release builds: null-pointer dereference -> SIGSEGV

This is reached via Mesh::EnsureNCMesh(true) for any valid mesh with a sufficiently high-valence vertex; such vertices arise naturally near sharp geometric features and material interfaces, so this is not specific to degenerate meshes.

Instead, this PR changes the cstruct to store the counters as int16_t (range +/-32767). The refcounts are never serialized, so the mesh file format is unaffected. int16_t preserves the compact Node memory layout (see commit eca90726): sizeof(Node) is unchanged (the two 16-bit counters fit in padding ahead of the 8-byte-aligned scale field), whereas widening to int would grow Node by 8 bytes (to 48 instead of 40). int16_t is also exactly 16 bits on every platform, removing the char signedness dependence that made the overflow threshold differ between architectures.

This PR also adds a regression test building a "spindle" of K tetrahedra sharing a central axis edge (axis vertices get valence K) for K in {255, 256, 512}, exercised through EnsureNCMesh, an order-1 H1 space, and a uniform refinement.

### CODE REVIEW

File: tests/unit/mesh/test_ncmesh.cpp [hunk 1]
```diff
@@ -2952,4 +2952,56 @@ TEST_CASE("InternalBoundaryProjectBdrCoefficient", "[NCMesh]")
...
+TEST_CASE("NCMesh high-valence vertex refcount", "[NCMesh]")
+{
+   // A vertex whose valence is a multiple of 256 (unsigned char) or >= 128
+   // (signed char) overflowed the 8-bit NCMesh::Node::vert_refc counter.
+   // HasVertex() then returned false, vertex_nodeId[] was left uninitialized,
+   // and NCMesh::OnMeshUpdated() dereferenced a null node -> assert/segfault.
+   //
+   // Build a "spindle" of K tetrahedra sharing the central axis edge so the
+   // two axis vertices each have valence K. K == 256 triggered the overflow;
+   // K == 255 did not.
```
Comment (hughcars) [inline tests/unit/mesh/test_ncmesh.cpp]: Only true for unsigned `char`; on x86 K=255 wraps to -1 and fails the same way. Just drop it.
```suggestion
   // two axis vertices each have valence K.
```

**Review by hughcars — APPROVED** [review]
Checked the layout with clang on arm64 and x86_64 (Darwin and Linux targets); `sizeof(Node)` and all field offsets are unchanged with `int16_t`:

| `real_t` | `char` | `int16_t` | `int` |
|---|---|---|---|
| double | 40 B, `scale@24` | 40 B, `scale@24` | 48 B, `scale@32` |
| float | 32 B, `scale@24` | 32 B, `scale@24` | 36 B, `scale@28` |

The two counters fill the 4 bytes at offset 12 between `Hashed2` and `vert_index`. One nit inline.
